auth/server/base/src/main/java/org/wildfly/security/auth/server/jwt/TokenProvider.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * JBoss, Home of Professional Open Source
+ *
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.auth.server.jwt;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+import org.wildfly.security.auth.server.SecurityIdentity;
+
+/**
+ * A service to issue JWT tokens for a given security identity
+ * and verify the contents of a token.
+ *
+ * @author <a href="mailto:szaldana@redhat.com">Sonia Zaldana</a>
+ */
+public interface TokenProvider {
+
+    /**
+     * Dynamically generates a JWT access token using the values in the security
+     * identity.
+     * @param securityIdentity the security identity
+     * @return a JWT access token in String representation
+     * @throws Exception
+     */
+    String issueAccessToken(SecurityIdentity securityIdentity) throws JwtException;
```

Comment:
Something we had discussed before was making sure that we could either generate our own tokens or make use of an external provider. One thing I'm wondering about is whether it makes sense for this interface method to take a `SecurityIdentity` as a parameter. The `SecurityIdentity` is definitely something that's need when generating our own tokens. But, it likely wouldn't be needed if using an external provider to obtain the token. In that case, an `OAuth2CredentialSource` would likely be used.